crates/dojo-lang/src/plugin_test.rs

```diff
@@ -101,18 +96,21 @@ pub fn test_expand_plugin_inner(
     // The path as to remain the same, because diagnostics contains the path
     // of the file. Which can cause error when checked without CAIRO_FIX=1.
     let tmp_dir = PathBuf::from("/tmp/plugin_test");
+
+    // starts from an empty dir to avoid conflicts with old/previous tests
+    let _ = std::fs::remove_dir_all(&tmp_dir);
```

Comment:
Couldn't we have some race here if tests are executed in parallel? In the original idea, the fact that they all use the same lib is not a problem.

And unfortunately, in the test framework we don't have access to the test name. It would have been better to actually separate all of them.

The main issue here with a true tmp dir is that, all the diagnostics in the generated files contains the path displayed under the span pointing at the error. For this reason, we need a stable path to have tests passing even without `CAIRO_FIX`.

Reply:
Ok so I removed this `std::fs::remove_dir_all` 👍 
On another hand, I also removed the `if` statements checking that`Scarb.toml` and `lib.cairo` already exist to be sure to override old/previous files.